seed/migrations/0125_dq_refactor.py

```diff
@@ -0,0 +1,98 @@
+# Generated by Django 2.2.10 on 2020-04-13 23:03
+
+from django.db import migrations, models
+
+
+def forwards(apps, schema_editor):
+    Rule = apps.get_model('seed', 'Rule')
+    # data_type not null:
+    #     data_type is 1:
+    #         text_match null: not_null -> not_null;
+    #                         !not_null -> required: required;
+    #                                   -> !required -> '';
+    #         text_match not null: include -> not_null: (+)not_null;
+    #                                      -> !not_null -> required: (+)required;
+    #     data_type is not 1 (0,2,3,4,5):
+    #         min and max null: not_null -> not_null;
+    #                         : !not_null -> required: required;
+    #                                     -> !required: '';
+    #         min or max not null: range -> not_null: (+)not_null;
+    #                                    -> !not_null -> (+)required: required;
+    # data_type null:
+    #     min and max null -> text_match null: not_null -> not_null;
+    #                                         !not_null -> required: required;
+    #                                                   -> !required: '';
+    #                      -> text_match not null: include -> not_null: (+)not_null;
+    #                                                      -> !not_null -> required: (+)required;
+    #     min or max not null: range -> not_null: (+)not_null;
+    #                                -> !not_null -> required: (+)required;
+
+    def create_rule(rule, condition, min, max, text_match):
+        Rule.objects.get_or_create(name=rule['name'], description=rule['description'], table_name=rule['table_name'],
+                                   field=rule['field'], enabled=rule['enabled'], data_type=rule['data_type'],
+                                   rule_type=rule['rule_type'], min=min, max=max, units=rule['units'],
+                                   data_quality_check_id=rule['data_quality_check_id'],
+                                   text_match=text_match, condition=condition)
```

Comment:
What's the reasoning for not disabling the fields like for not_null?

Also - what do you think about the the DB migration method being changed as suggested to avoid essentially duplicate Rules?

Reply:
So my understanding for 'required' for now is this could be for potential column mapping which to me works as a place holder and is allowed to include nullable values. There are records found where 'required' is true but 'not_null' is false and some of these rule_fields are 'notes'. If we disabled required, it'll be no difference from not_null. The only difference is it will not be reported if it receives a null value.

Reply:
> Also - what do you think about the the DB migration method being changed as suggested to avoid essentially duplicate Rules?
It makes sense to me. I do want to confirm with you that these are the essential elements for properly creating additional rules without being duplicated. Here i used all elements except for severity level and label id as these two definitely create duplicates.

Reply:
> Also - what do you think about the the DB migration method being changed as suggested to avoid essentially duplicate Rules?

It makes sense to me. I do want to confirm with you that these are the essential elements for properly creating additional rules without being duplicated. Here i used all elements except for severity level and label id as these two definitely create duplicates.

Reply:
Per our call, 
- min, max, and text_match will be disabled on the frontend when Rule has condition "Required".
- min, max, and text_match will be set to None for newly created Rules with condition 'required' or 'not_null'.
- bring back "required" check on None values as it was in develop. Any potential changes to the underlying required logic will be made in a separate PR.